charts/cadc-tap/values.yaml

```diff
@@ -2,83 +2,152 @@
-  enabled: false
+# Settings for the ingress rules.
+ingress:
+  # -- Hostname for the ingress
+  # @default -- None, must be set
+  host: ""
 
-  # Path to load secrets for in the vault.
-  path: ""
+  # -- Gafaelfawr auth query string
+  gafaelfawrAuthQuery: "scope=read:tap&auth_type=basic&delegate_to=tap"
 
-secrets:
-  # Create a normal k8s secret using the values provided here.
-  # Even if you don't have these secrets, you still need to have
-  # either secrets.enabled or vault_secrets.enabled.
-  enabled: true
+  # -- Additional annotations to use for endpoints that allow anonymous
+  # access, such as `/capabilities` and `/availability`
+  anonymousAnnotations: {}
 
-  # GCE service account credential JSON file, base64 encoded.
-  # You can also pass this in using --set-file secrets.google_creds=google_creds.json
-  # on the command line, to allow not checking it in.
-  google_creds: ""
+  # -- Additional annotations to use for endpoints that are authenticated,
+  # such as `/sync`, `/async`, and `/tables`
+  authenticatedNnnotations: {}
 
-# Settings for the ingress rules.
-ingress:
-  # Force SSL, which is used in annotations and rewriting redirects.
-  ssl: true
-  # Additional annotations to use for endpoints that allow anonymous
-  # access, such as /capabilities, /availability
-  anonymous_annotations: {}
-  # Additional annotations to use for endpoints that are authenticated,
-  # such as /sync, /async, /tables
-  authenticated_annotations: {}
+networkPolicy:
+  # -- Whether to restrict access to the service. Only enable if the ingress
+  # controller namespace is tagged with `gafaelfawr.lsst.io/ingress: "true"`.
+  enabled: false
 
+# -- Resource limits and requests for the Gafaelfawr frontend pod
 resources: {}
-  # We usually recommend not to specify default resources and to leave this as a conscious
-  # choice for the user. This also increases chances charts run on environments with little
-  # resources, such as Minikube. If you do want to specify resources, uncomment the following
-  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
-  # limits:
-  #  cpu: 100m
-  #  memory: 128Mi
-  # requests:
-  #  cpu: 100m
-  #  memory: 128Mi
-
-# Setting for the java heap size, which will set the maximum size of the heap.  Otherwise
-# java will determine it based on how much memory is available and black maths.
-jvm_max_heap_size: 4G
-
-aux_resources: {}
-  # Same for resources, but for the non-main TAP service services (such as databases).
 
+# -- Annotations for the Gafaelfawr frontend pod
+podAnnotations: {}
+
+# -- Node selector rules for the Gafaelfawr frontend pod
 nodeSelector: {}
 
+# -- Tolerations for the Gafaelfawr frontend pod
 tolerations: []
 
+# -- Affinity rules for the Gafaelfawr frontend pod
 affinity: {}
+
+# -- Path to the Vault secret (`secret/k8s_operator/<host>/tap`, for example)
+# @default -- None, must be set
+vaultSecretsPath: ""
```

Comment:
Following the Phalanx docs, I usually set the VaultSecretsPath in the Phalanx service along with the VaultSecret. I think it is better just to assume a k8s secret in the chart.

Reply:
This is probably a good conversation for stand-up or co-work. Right now, essentially all of our charts use this approach with `vaultSecretsPath`; I'm not adverse to changing it, but we should be consistent, so we should change it for all of them at once if we do, I think.

The concern I would have about relying on a secret is that the chart then depends on an external resource that the chart itself doesn't provide, which I think might be confusing. I know some third-party charts work that way, and it's to some extent unavoidable for a general chart because everyone handles secrets differently. But we always use Vault, and this approach means that installing the chart will fail if Vault isn't configured properly and otherwise will always work, which I think is a little bit better. But I could be missing an argument in favor of the separate secret approach.

I'm going to move forward with this approach for now to make it match the other charts, but will go back and change this and the others if we decide on the secret approach.

Reply:
Thanks for the clarification, no need change the way we are doing I'll just follow this convention.